packages/engine/src/simulation/package/state/packages/behavior_execution/tasks.rs

```diff
@@ -43,6 +43,19 @@ impl WorkerHandler for ExecuteBehaviorsTask {
             payload: task_msg.into(),
         })
     }
+
+    fn combine_task_messages(&self, mut task_messages: Vec<TaskMessage>) -> Result<TaskMessage> {
+        if task_messages.is_empty() {
+            Err(Error::Unique(
+                "Expected there to be at least one TaskMessage returned by the \
+                 BehaviorExecutionTask"
+                    .to_string(),
+            ))
+        } else {
+            // The `ExecuteBehaviorsTaskMessage` is empty so there's no special combining logic
+            Ok(task_messages.remove(0))
+        }
+    }
```

Comment:
I think `combine_messages` and `combine_task_messages` should actually do the same thing, so I would extract a function that they both call

Reply:
Yep, agreed, have done that